Service/SamlDTO.php

```diff
@@ -118,14 +120,161 @@ public function __construct(array $samlAttributes)
             if (0 === $value) {
                 throw new AakSamlException(sprintf('Invalid id value "%s" - Cannot convert to integer.', $value));
             }
-        });
-
-        /* @var int[] $ids */
-        $this->companyId = $ids[0]; // @phpstan-ignore assign.propertyType
-        $this->divisionId = $ids[1]; // @phpstan-ignore assign.propertyType
-        $this->departmentId = $ids[2]; // @phpstan-ignore assign.propertyType
-        $this->subDepartmentId = $ids[3]; // @phpstan-ignore assign.propertyType
-        $this->officeId = $ids[4]; // @phpstan-ignore assign.propertyType
+
+            $departmentIds[] = $value;
+        }
+
+        $this->departmentIds = $departmentIds;
+
+        if (0 === count($this->departmentIds)) {
+            throw new AakSamlException(sprintf('No organization ids given in claims: "%s"', $this->getAttributeValue(self::ID_ARRAY_ATTRIBUTE, $samlAttributes)));
+        }
+
+        // We can map claims from level 1-5. If we see id's for deeper levels we don't know the corresponding claim for
+        // department name.
+        $this->companyId = array_key_exists(0, $this->departmentIds) ? $this->departmentIds[0] : null;
+        $this->divisionId = array_key_exists(1, $this->departmentIds) ? $this->departmentIds[1] : null;
+        $this->departmentId = array_key_exists(2, $this->departmentIds) ? $this->departmentIds[2] : null;
+        $this->subDepartmentId = array_key_exists(3, $this->departmentIds) ? $this->departmentIds[3] : null;
+        $this->officeId = array_key_exists(4, $this->departmentIds) ? $this->departmentIds[4] : null;
```

Comment:
Any specific reason for not writing this as

```suggestion
        $this->companyId = $this->departmentIds[0] ?? null;
        $this->divisionId = $this->departmentIds[1] ?? null;
        $this->departmentId = $this->departmentIds[2] ?? null;
        $this->subDepartmentId = $this->departmentIds[3] ?? null;
        $this->officeId = $this->departmentIds[4] ?? null;
```

?